hooks/index.ts

```diff
@@ -127,10 +128,20 @@ export const useDataSourceContext = () => {
 
   useEffect(() => {
     if (view?.dataSourceId)
-      dispatch(setDataSourceId(view.dataSourceId.toString()));
+    dispatch(setDataSourceId(view.dataSourceId.toString()));
     if (view?.tableName) dispatch(setTableName(view.tableName));
   }, [view]);
 
+  const dashboardId = router.query.dashboardId as string;
+  const { dashboard } = useDashboardResponse(dashboardId);
+
+  useEffect(() => {
+    if (dashboard?.dataSourceId) {
+      dispatch(setDataSourceId(dashboard.dataSourceId.toString()));
+      dispatch(setTableName(""));
```

Comment:
Maybe we need to invalidate the view too?

Reply:
not necessary